homeassistant/helpers/event.py

```diff
@@ -1389,6 +1389,45 @@ def utc_converter(utc_now: datetime) -> None:
 track_point_in_time = threaded_listener_factory(async_track_point_in_time)
 
 
+@dataclass(slots=True)
+class _TrackPointUTCTime:
+    hass: HomeAssistant
+    job: HassJob[[datetime], Coroutine[Any, Any, None] | None]
+    utc_point_in_time: datetime
+    expected_fire_timestamp: float
+    _cancel_callback: asyncio.TimerHandle | None = None
+
+    def __post_init__(self) -> None:
```

Comment:
This is different from SunListener in that it automatically attaches inside the constructor. Having such side effects feels a bit dirty. Can we do an `def async_attach` like the sun?

Reply:
Sure it would be a simple rename of the function + call

Reply:
8d1c3f4fb8355c1866f040e4958080b40465db64

Reply:
I went with `start` instead of `async_attach` as it seemed like a better name for what it does. happy to adjust otherwise though

Reply:
changed my mind on that. I think there is more value in being consistant so I went with `async_attach`